eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/FixImagesVarDirCommand.php

```diff
@@ -0,0 +1,416 @@
+    if (0 === error_reporting()) {
+        return false;
+    }
+
+    throw new RuntimeException($errstr, 0);
+});
+
+class FixImagesVarDirCommand extends ContainerAwareCommand
+{
+    const DEFAULT_ITERATION_COUNT = 100;
+    const STORAGE_IMAGES_PATH = '/storage/images/';
+
+    /**
+     * @var int
+     */
+    protected $done = 0;
+
+    /**
+     * @var string
+     */
+    private $phpPath;
+
+    /**
+     * @var bool
+     */
+    private $dryRun;
+
+    /**
+     * @var bool
+     */
+    private $moveFiles;
+
+    /**
+     * @var \Doctrine\DBAL\Connection
+     */
+    private $connection;
+
+    /**
+     * @var int
+     */
+    private $varDir;
+
+    /**
+     * @var array
+     */
+    private $imageAttributes = [];
+
+    protected function configure()
+    {
+        $this
+            ->setName('ezplatform:fix_images_var_dir')
+            ->setDescription(
+                'This update script will fix references to images that aren\'t part of the current var_dir.'
+            )
+            ->addOption(
+                'dry-run',
+                null,
+                InputOption::VALUE_NONE,
+                'Execute a dry run'
+            )
+            ->addOption(
+                'move-files',
+                null,
+                InputOption::VALUE_NONE,
+                'Move image files between var directories'
+            )
+            ->addOption(
+                'iteration-count',
+                null,
+                InputArgument::OPTIONAL,
+                'Limit how much records get updated by single process',
+                self::DEFAULT_ITERATION_COUNT
+            )
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> fixes referefences to images that aren't part of the current var_dir.
+
+This may for instance occur when the var_dir setting is changed. This script will rename the files, and update the
+database references to the new path
+
+Since this script can potentially run for a very long time, to avoid memory exhaustion run it in
+production environment using <info>--env=prod</info> switch.
+
+EOT
+            );
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        /* @var \Doctrine\DBAL\Connection $databaseHandler */
+        $this->connection = $this->getContainer()->get('ezpublish.api.search_engine.legacy.connection');
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int|void|null
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $iterationCount = (int) $input->getOption('iteration-count');
+        $this->dryRun = $input->getOption('dry-run');
+        $this->moveFiles = $input->getOption('move-files');
+        $consoleScript = $_SERVER['argv'][0];
```

Comment:
Isn't it better to do `$input->getArgument` than mess with `$_SERVER`?
https://symfony.com/doc/3.4/console/input.html

Reply:
`$_SERVER['argv'][0];` is for determining if script is run by `app/console` or `bin/console`. I'm not sure if there is a prettier way of doing this.

Reply:
Right! But will it work both if you run `php bin/console` and with just `bin/console`?

Reply:
Yes it will, nevermind :)